ext/mbstring/libmbfl/mbfl/mbfl_encoding.h

```diff
@@ -285,6 +285,7 @@ static inline void mb_convert_buf_reset(mb_convert_buf *buf, size_t len)
 }
 
 MBFLAPI extern const mbfl_encoding *mbfl_name2encoding(const char *name);
+MBFLAPI extern const mbfl_encoding *mbfl_name2encoding_ex(const char *name, size_t name_len);
```

Comment:
Yeah, there are places where we can pass `ZSTR_LEN(something)`. Although that would be a behaviour change because those places seem to allow \0 in strings (the function was never able to handle \0 in strings and silently cut off the string there, but passing ZSTR_LEN would thus change the \0 behaviour). I see you're making a PR to refactor some stuff anyway, so feel free to take a look at those places :)